### PR TITLE
feat: add metadata for legacy and complicated resource examples

### PR DESCRIPTION
**Pull request by Sita04**
Adding Metadata for:
1. Os Login
2. Create disk snapshot
3. List log entries

### CODE REVIEW

File: products/compute/metadata/oslogin_service_account_ssh.toml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# Also, instead of service account keys, can we use impersonation?
+commands = [
+            "gcloud iam service-accounts create $SERVICE_ACCOUNT",
+            "gcloud iam service-accounts add-iam-policy-binding  $SERVICE_ACCOUNT --member='serviceAccount:$SERVICE_ACCOUNT --role='roles/serviceAccountsUser'",
+            "gcloud compute instances create $INSTANCE_NAME --image-family=ubuntu-2204-lts --image-project=ubuntu-os-cloud --boot-disk-size=\"25GB\" --boot-disk-type=\"pd-standard\" --boot-disk-auto-delete=\"true\" --network=\"default\" --machine-type=\"e2-micro\" --tags=\"ssh-oslogin-test\" --network-tier=\"PREMIUM\" --service-account=$SERVICE_ACCOUNT --scopes=\"https://www.googleapis.com/auth/cloud-platform\" --metadata=\"enable-oslogin=TRUE\" --zone=\"us-central1-a\"",
+            "gcloud compute instances add-iam-policy-binding $INSTANCE_NAME --zone=$ZONE --member=serviceAccount:$SERVICE_ACCOUNT --role='roles/compute.osLogin'",
+            "gcloud compute firewall-rules create $FIREWALL_RULE_NAME --network=\"default\" --target-tags=\"ssh-oslogin-test\" --source-ranges=\"0.0.0.0/0\" --allow tcp:22"
+]
```
Comment (m-strzelczyk) [inline products/compute/metadata/oslogin_service_account_ssh.toml:82]: I think we will need to add some kind of `sleep 30` to let this all settle down before starting the sample.

**Review by m-strzelczyk — COMMENTED** [review]
General idea I want to brainstorm:

I'm think we should remove PROJECT_ID from the variables configuration. Project config is special and deserves some different treatment, because:
 * It is required for every sample, no matter the product
 * Different users might want to use different projects
 * Users who want to run tests manually will want to just use their "default" projects easily, without updating dozens of TOML files.

That's why I want to suggest the $PROJECT_ID value should be always initialized using `google.auth.default()` or grabbed from an env value like `$GOOGLE_CLOUD_PROJECT`. Everyone will be able to switch between projects easily then, without modifying config files.

**Review by m-strzelczyk — APPROVED** [review]
Approving so you can merge when you think we're done.

File: products/logging/metadata/list_log_entries.toml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+[tests.setup]
+# This needs to exit with code 0. If it doesn't the test will fail.
+# If shared_resources = false, this command will be executed for every language separately.
+# If it's true, the command will be executed only once before all the tests.
+commands = ["gcloud logging write $LOGGER_NAME \"$LOG_MESSAGE\" "]
+
+[tests.check]
+# Check if the log_message is present in the returned log entries.
+command = "gcloud logging logs list | grep $LOG_MESSAGE"
```
[the remark this answers is not in the document]
Comment (Sita04) [reply products/logging/metadata/list_log_entries.toml:61]: That's a nice catch.
Makes sense. I've updated the file.

File: products/compute/metadata/oslogin_service_account_ssh.toml (hunk 1 above)
Comment (Sita04) [reply products/compute/metadata/oslogin_service_account_ssh.toml:82]: Agreed! Few of these are LRO. Will update the commands to include sleep timer.

**Comment by Sita04** [comment]
> General idea I want to brainstorm:
> 
> I'm think we should remove PROJECT_ID from the variables configuration. Project config is special and deserves some different treatment, because:
> 
> * It is required for every sample, no matter the product
> * Different users might want to use different projects
> * Users who want to run tests manually will want to just use their "default" projects easily, without updating dozens of TOML files.
> 
> That's why I want to suggest the $PROJECT_ID value should be always initialized using `google.auth.default()` or grabbed from an env value like `$GOOGLE_CLOUD_PROJECT`. Everyone will be able to switch between projects easily then, without modifying config files.

Good points. We definitely don't want to manually update those variables for every run.
But we still have to grab the value from the environment variable to be passed to the [[tests]].

I'll add a TODO for this when we decide on the approach and get back.

**Pull request merged**